### PR TITLE
Complete part of lease,and move functional still working on it

### PR DESCRIPTION
**Pull request by Csedge**

### CODE REVIEW

File: src/main/java/assets/game.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ var vertical;
 var is_sonar=0;
 var remain_Sonar=2;
 var allow_sonar=0;
+var allow_leaser=0;
```
Comment (hcmlax) [inline src/main/java/assets/game.js]: Would a laser class be more efficient? we could then add the class to the board class and use it through that

File: src/main/java/cs361/battleships/models/Game.java [hunk 2]
```diff
@@ -35,11 +35,15 @@ public boolean Sonar_Pluse(int x,char y){
...
+    public boolean Move_ship(int type){
+        playersBoard.move_ship(type);
+        return true;
+    }
     /*
 	DO NOT change the signature of this method. It is used by the grading scripts.
 	 */
-    public boolean attack(int x, char  y) {
-        Result playerAttack = opponentsBoard.attack(x, y);
+    public boolean attack(int x, char  y,int is_leaser) {
```
Comment (hcmlax) [inline src/main/java/cs361/battleships/models/Game.java]: How are we going to check if there are multiple ships that are in the same location? when the attack function is called, it only grabs one ship if it is hit

File: src/main/java/views/ApplicationController/index.ftl.html [hunk 3]
```diff
@@ -18,11 +18,17 @@ <h2 id="playerboard"> player board</h2>
                     <div class="shipTypes" style="width:100%"><button id="place_submarine" class="btn btn-info">&nbsp;&nbsp;&nbsp;Place Submarine&nbsp;&nbsp;&nbsp;</button></div><br/>
                     <div class="sonar hidden"><button id="sonar_button" class="disabled"><img class="sonarIcon" src="/assets/sonar-icon-8.jpg" alt="sonarIcon"><span>Sonar </span></button></div>
                 </div>
-
+            <div class="ship_buttons">
```
Comment (hcmlax) [inline src/main/java/views/ApplicationController/index.ftl.html]: Should we add a general move ship button, then give the user the options to pick a direction after they click it?

File: src/test/java/cs361/battleships/models/BoardTest.java [hunk 4]
```diff
@@ -50,15 +50,15 @@ public void testAttackSameSquareMultipleTimes() {
         Ship minesweeper = new Minesweeper();
         board.placeShip(minesweeper, 1, 'A', true);
         //board.attack(2, 'A');
-        Result result = board.attack(2, 'A');
+        Result result = board.attack(2, 'A',0);
```
Comment (hcmlax) [inline src/test/java/cs361/battleships/models/BoardTest.java]: Thanks for updating the tests to work with the new additions, I will implement some for the submarine class as well

File: src/main/java/cs361/battleships/models/Board.java [hunk 5]
```diff
@@ -42,6 +42,27 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 		placedShip.placeCaptainsQuarters();
 		return true;
 	}
+	public boolean move_ship(int type){
```
Comment (hcmlax) [inline src/main/java/cs361/battleships/models/Board.java]: This looks like a good template to implement the move ship feature.  Adding a test that confirms that a ship has changed its location will be a good way to make sure this is working as intended

File: src/main/java/assets/game.js (hunk 1 above)
Comment (rise3n) [reply src/main/java/assets/game.js]: Instead it should be something like weapon class, and laser class would be the subclass then if there are more weapons in the future.

**Pull request merged**